src/databricks/labs/ucx/hive_metastore/federation.py

```diff
@@ -0,0 +1,116 @@
+import collections
+import logging
+from enum import Enum
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import AlreadyExists, NotFound, BadRequest
+from databricks.sdk.service.catalog import (
+    ConnectionInfo,
+    SecurableType,
+    Privilege,
+    PermissionsChange,
+)
+
+from databricks.labs.ucx.account.workspaces import WorkspaceInfo
+from databricks.labs.ucx.hive_metastore import ExternalLocations
+
+
+logger = logging.getLogger(__name__)
+
+
+class ConnectionType(Enum):
```

Comment:
ah, that's the hack you mentioned - we can live with that.

Reply:
We will reintroduce when we bring back federation